hermes/crates/cardano-chain-follower/src/follow.rs

```diff
@@ -0,0 +1,577 @@
+};
+
+/// Default [`Follower`] block buffer size.
+const DEFAULT_CHAIN_UPDATE_BUFFER_SIZE: usize = 32;
+
+/// Enum of chain updates received by the follower.
+pub enum ChainUpdate {
+    /// New block inserted on chain.
+    Block(MultiEraBlockData),
+    /// Chain rollback to the given block.
+    Rollback(MultiEraBlockData),
+}
+
+impl ChainUpdate {
+    /// Gets the chain update's block data.
+    #[must_use]
+    pub fn block_data(&self) -> &MultiEraBlockData {
+        match self {
+            ChainUpdate::Block(block_data) | ChainUpdate::Rollback(block_data) => block_data,
+        }
+    }
+}
+
+/// Builder used to create [`FollowerConfig`]s.
+pub struct FollowerConfigBuilder {
+    /// Block buffer size option.
+    chain_update_buffer_size: usize,
+    /// Where to start following from.
+    follow_from: PointOrTip,
+    /// Path to the Mithril snapshot the follower should use.
+    mithril_snapshot_path: Option<PathBuf>,
+}
+
+impl Default for FollowerConfigBuilder {
+    fn default() -> Self {
+        Self {
+            chain_update_buffer_size: DEFAULT_CHAIN_UPDATE_BUFFER_SIZE,
+            follow_from: PointOrTip::Tip,
+            mithril_snapshot_path: None,
+        }
+    }
+}
+
+impl FollowerConfigBuilder {
+    /// Sets the size of the chain updates buffer used by the [`Follower`].
+    ///
+    /// # Arguments
+    ///
+    /// * `chain_update_buffer_size`: Size of the chain updates buffer.
+    #[must_use]
+    pub fn chain_update_buffer_size(mut self, block_buffer_size: usize) -> Self {
+        self.chain_update_buffer_size = block_buffer_size;
+        self
+    }
+
+    /// Sets the point at which the follower will start following from.
+    ///
+    /// # Arguments
+    ///
+    /// * `from`: Sync starting point.
+    #[must_use]
+    pub fn follow_from<P>(mut self, from: P) -> Self
+    where P: Into<PointOrTip> {
+        self.follow_from = from.into();
+        self
+    }
+
+    /// Sets the path of the Mithril snapshot the [`Follower`] will use.
+    ///
+    /// # Arguments
+    ///
+    /// * `path`: Mithril snapshot path.
+    #[must_use]
+    pub fn mithril_snapshot_path(mut self, path: PathBuf) -> Self {
+        self.mithril_snapshot_path = Some(path);
+        self
+    }
+
+    /// Builds a [`FollowerConfig`].
+    #[must_use]
+    pub fn build(self) -> FollowerConfig {
+        FollowerConfig {
+            chain_update_buffer_size: self.chain_update_buffer_size,
+            follow_from: self.follow_from,
+            mithril_snapshot_path: self.mithril_snapshot_path,
+        }
+    }
+}
+
+/// Configuration for the Cardano chain follower.
+pub struct FollowerConfig {
+    /// Configured chain update buffer size.
+    pub chain_update_buffer_size: usize,
+    /// Where to start following from.
+    pub follow_from: PointOrTip,
+    /// Path to the Mithril snapshot the follower should use.
+    pub mithril_snapshot_path: Option<PathBuf>,
+}
+
+/// Cardano chain follower.
```

Comment:
I think we should put here more description about why to use `Follower`, so it should somehow explains what the difference with the `Reader`. Because right now for someone who does not have a context how it works does not obvious.